code/modules/medical/genetics/geneticsScanner.dm

```diff
@@ -264,12 +272,20 @@ var/list/genescanner_addresses = list()
 			return 1
 		..()
 
+	proc/update_occupant()
+		var/mob/living/carbon/human/H = src.occupant
+		if (istype(H))
+			src.occupant_icon = character_preview_icon(H.bioHolder.mobAppearance, H.mutantrace)
```

Comment:
Sanity check on how often this line gets called. This uses `getFlatIcon` under the hood, which isn't particularly cheap. AFAICT `update_occupant` is called whenever the `editappearance` action is sent, which seems to be done whenever any of those controls are messed with. Big concern for me is dragging those dials around slowly causing *lots* of calls here.

Reply:
The dials only send the signal back to the server when you release them. Not really much more efficient I can make it unless I render the character sprite on the client side somehow.

Reply:
No way, current character setup lags the server a lot when someone decides to edit their profiles during the round. Replace this with something else.

Reply:
@BenLubar I'm going to have to dabble with working out how to show changing sprites (for this and for an inevitable rework of the Character Setup popup) efficiently should work. I am aware that there's a "ByondUi" component that might be useful here, you're welcome to take a look into that if you'd like, else I'll add to my to-do list.

If you do look into it, please try to make something we can re-use for showing a character sprite rather than baking into the genetics console stuff.

Reply:
Have you considered changing the character preview a map window instead? we did it on /tg/ and it greatly reduce lag from character setup

Reply:
If someone can do that, be my guest!

I have no idea how the map window stuff works in byond after looking at it quite a bit.

Reply:
> If someone can do that, be my guest!
> 
> I have no idea how the map window stuff works in byond after looking at it quite a bit.

TG's character setup has an example.